### DISCUSSION TITLE
项目使用了 dumi 来做文档，项目里用的 corejs@2.x，umi 里却锁了 corejs@3.x，导致项目文档启动时候报错

### DISCUSSION THREAD

**Discussion by jo0ger** [umijs/umi#12129] [Q&A]
[core-js@3.x](mailto:core-js@3.x) 包含了 flat/web-url/web-url-search-params
[core-js@2.x](mailto:core-js@2.x) 没有 web-url ，flat 对应的 polyfill 仍然还是 flatten

导致启动时候找不到对应的 core-js 里的包

<img width="1342" alt="image" src="https://github.com/umijs/umi/assets/16385416/ca23b4cd-7548-4230-8618-d91052dac6ff">

**Comment by fz6m** [comment]
umi 自带一个 babel 插件，这个插件会锁定所有 core js 的导入到唯一的内置 core js ，这个版本是 umi 依赖本身附带并决定的。

如上文所述，因为最终只有唯一的一个 core js ，不明白你说的 core js 2 和 3 是哪里来的。

如有问题，需要给一个最小复现。

若认为 dumi 存在问题，可以到 dumi 仓库反馈看看。

**Reply by jo0ger** [reply]
corejs 2 是我项目其他打包工具依赖的，现在我想只用 dumi 和 umi 做文档搭建，打包还是用我自己的工具

然后我的打包工具有一些 runtime 的包，也依赖了 corejs2，我的业务代码里又 import 了 这些 runtime

然后我用 dumi 在本地做文档 doc 开发的时候，就报错了

**Reply by fz6m** [reply]
不同的 corejs 版本中的 polyfill 可能会发生冲突，由于 umi 项目本身就有 corejs 3 的 polyfill 了，所以你的 runtime 不再需要导入 polyfill ，考虑将你自己的 runtime 中的 corejs 升级为 3 一致或者打包一份没有 polyfill 的产物给 umi 使用。
